deps/amqp10_client/src/amqp10_client_session.erl

```diff
@@ -771,6 +778,17 @@ filter_value_type({T, _} = V) when is_atom(T) ->
     %% looks like an already tagged type, just pass it through
     V.
 
+translate_terminus_capabilities(Capabilities) when is_binary(Capabilities) ->
+    {symbol, Capabilities};
+translate_terminus_capabilities(CapabilitiesList) when is_list(CapabilitiesList) ->
+    {array, symbol, [filter_capability(V) || V <- CapabilitiesList]}.
+
+filter_capability(V) when is_binary(V) ->
+    {symbol, V};
+filter_capability(_) ->
+    %% Any other type is ignored
+    {}.
```

Comment:
Is this code tested?
I'm pretty sure this tuple with 0 elements will crash later in `amqp10_binary_generator`.
See my comment in https://github.com/rabbitmq/rabbitmq-server/pull/11337#discussion_r1647195471